### ISSUE TITLE
FPN-23 | Phase 5 | Ablation Study: Performance Gain from Feedback Incorporation

### ISSUE THREAD

**Issue by LavanyaSrinivas06** [LavanyaSrinivas06/AI-First-Preauth-Fraud-Detection#72]
### Ticket ID: FPN-23  
Phase: Phase 5 — Retraining & Continuous Learning  
Branch Name: feature/FPN-23-ablation-feedback-gain

---

### Context
Ablation experiments quantify how analyst feedback improves model performance over baseline.  
This is critical for validating the adaptive learning component in your thesis — showing measurable benefit from human-in-the-loop retraining.

---

### Scope
- Compare baseline (pre-feedback) and retrained (post-feedback) supervised models.  
- Evaluate both on the same test and feedback holdout sets.  
- Quantify gain in F1, Precision, Recall, and ROC-AUC.  
- Visualize performance deltas and write a structured report.

---

### Steps

1. **Prepare datasets**
   - Use:
     - Original test set: `data/processed/test.parquet`
     - Feedback-derived holdout: latest portion of `data/feedback.parquet` not used in retraining.
   - Ensure identical preprocessing pipeline is used for both models.

2. **Model loading**
   - Load:
     - `artifacts/models/xgb/baseline/xgb_model.pkl`
     - `artifacts/models/xgb/<latest>/xgb_model.pkl`
   - Maintain same inference path as in API service.

3. **Evaluation metrics**
   - Compute per-model metrics:
     - Precision, Recall, F1, ROC-AUC, PR-AUC
   - Evaluate on both test and feedback holdout sets.
   - Store results in a comparison table:
     ```
     metric | baseline_test | retrained_test | baseline_feedback | retrained_feedback | gain (%)
     ```

4. **Visualizations**
   - Generate:
     - Bar charts for metric improvement
     - ROC and PR curves (both models)
     - Confusion matrix deltas
   - Save plots under:
     ```
     artifacts/plots/ablation/
     ```

5. **Statistical validation (optional but recommended)**
   - Apply McNemar’s test or bootstrap resampling to verify statistical significance of improvements.

6. **Documentation**
   - Create `docs/ablation_study_feedback.md`:
     - Experimental setup (datasets, parameters, versions)
     - Metric tables and plots
     - Observations on where retraining helps (recall gains, fewer false positives)
     - Discussion on trade-offs (e.g., recall vs precision)
     - Note reproducibility command and artifact versions.

7. **Reproducibility**
   - Script: `scripts/run_ablation_feedback.py`
     - Loads both models, runs evaluation, saves tables/plots automatically.

---

### Acceptance Criteria
- [ ] Both baseline and retrained models evaluated on test and feedback holdout sets.  
- [ ] Comparison table with metrics and gain percentages generated.  
- [ ] Plots (ROC, PR, confusion matrix) saved under `artifacts/plots/ablation/`.  
- [ ] `docs/ablation_study_feedback.md` includes tables, plots, and discussion.  
- [ ] Script reproducibly generates the full analysis.

---

### Artifacts
- `scripts/run_ablation_feedback.py`
- `artifacts/plots/ablation/`
- `docs/ablation_study_feedback.md`

---

### Helpful Links
- McNemar’s test for model comparison: https://machinelearningmastery.com/mcnemars-test-for-machine-learning/
- sklearn metrics: https://scikit-learn.org/stable/modules/model_evaluation.html

---

### Effort Estimate
M (1–2 days)
